### PR TITLE
[CPM-90]Add plan routes

### PR DESCRIPTION
**Pull request by daltonlim**

### CODE REVIEW

File: api/src/routes/plan-route.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    Plan.findOneAndDelete({ name: req.params.name }, (err, plan) => {
+        if (err) {
+            LOGGER.error(err);
+            res.status(400).json({ msg: err.message });
+        } else {
+            if (plan === null) {
+                LOGGER.error("Plan does not exist");
+                res.status(404).json({ msg: "Requested object not found" });
+            } else {
+                LOGGER.info("DELETE request succeeded for /Plan/:name");
```
Comment (HongShi10) [inline api/src/routes/plan-route.js:202]: change :name to actually send the name so do something like `${req.params.name}`

**Review by HongShi10 — CHANGES_REQUESTED** [review]

File: api/src/routes/plan-route.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        if (err) {
+            LOGGER.error(err);
+            res.status(400).json({ msg: err.message });
+        } else {
+            if (plans.length <= 0) {
+                LOGGER.info("No plans found");
+                res.status(404).json();
+            } else {
+                LOGGER.info("GET request succeeded for /plan");
+                LOGGER.info(plans);
```
Comment (Dinith1) [inline api/src/routes/plan-route.js:85]: Maybe only log the data in debug mode

**Review by Dinith1 — APPROVED** [review]
Left a comment but LGTM

File: api/src/routes/plan-route.js (hunk 2 above)
Comment (daltonlim) [reply api/src/routes/plan-route.js:85]: how to do

File: api/src/routes/plan-route.js (hunk 1 above)
Comment (daltonlim) [reply api/src/routes/plan-route.js]: Fixed

File: api/src/routes/plan-route.js (hunk 2 above)
Comment (daltonlim) [reply api/src/routes/plan-route.js]: done?

**Review by wryao64 — APPROVED** [review]
LGTM - thanks for removing the unnecessary console.logs

**Pull request merged**